src/components/ec-metroline/ec-metroline.spec.js

```diff
@@ -0,0 +1,369 @@
+        data-test="footer-cta-button"
+      >
+        Continue
+      </button>
+    </template>
+  </ec-metroline-item>
+
+  <ec-metroline-item
+    :id="2"
+  >
+    <template #heading>
+      <span>Item 2 Heading</span>
+    </template>
+
+    <template #sub-heading>
+      <span>Item 2 Sub-heading</span>
+    </template>
+
+    <template #header-cta="{ goTo }">
+      <button
+        @click="goTo"
+        data-test="header-cta-button"
+      >
+        Edit
+      </button>
+      <p>Item 2 Main Content</p>
+    </template>
+
+    <template #main="{ status }">
+    </template>
+
+    <template #footer-cta="{ goToNext }">
+      <button
+        @click="goToNext"
+        data-test="footer-cta-button"
+      >
+          Continue
+      </button>
+    </template>
+  </ec-metroline-item>
+</ec-metroline>
+`;
+
+describe('EcMetroline', () => {
+  it('should render a metroline with items', async () => {
+    const wrapper = await mountMetrolineAsTemplate(metrolineWithItemsTemplate);
+
+    expect(wrapper.element).toMatchSnapshot();
+  });
+
+  describe(':props', () => {
+    it('should render the metroline badgeText', async () => {
+      const wrapper = await mountMetrolineAsTemplate(
+        `<ec-metroline>
+          <ec-metroline-item :id="1" badgeText="IV">
+            <template #heading>
+              <span>Item 1 Heading</span>
+            </template>
+
+            <template #main="{ status }">
+              <p>Item 1 Main Content</p>
+            </template>
+          </ec-metroline-item>
+        </ec-metroline>`,
+      );
+
+      expect(wrapper.element).toMatchSnapshot();
+    });
+
+    it('should throw an error if we don\'t pass an id', () => {
+      withMockedConsole((errorSpy) => {
+        mountMetrolineAsTemplate(
+          `<ec-metroline>
+            <ec-metroline-item>
+              <template #heading>
+                <span>Item 1 Heading</span>
+              </template>
+            </ec-metroline-item>
+          </ec-metroline>`,
+        );
+
+        expect(errorSpy.mock.calls[0][0]).toContain('Missing required prop: "id"');
+        expect(errorSpy).toHaveBeenCalledTimes(1);
+      });
+    });
+  });
+
+  describe('@events', () => {
+    it('should emit a "change" event when we go to next item', async () => {
+      const onChange = jest.fn();
+      const wrapper = await mountMetrolineAsTemplate(`
+      <ec-metroline @change="onChange">
+        <ec-metroline-item
+          v-for="index in 2"
+          :key="index"
+          :id="index">
+          <template #heading>
+            <span>Heading</span>
+          </template>
+
+          <template #footer-cta="{ goToNext }">
+            <button
+              data-test="footer-cta-button"
+              @click="goToNext"
+            >
+              Continue
+            </button>
+          </template>
+        </ec-metroline-item>
+      </ec-metroline>`,
+      {
+        methods: {
+          onChange,
+        },
+      });
+
+      wrapper
```

Comment:
you can await the trigger('click') and drop the `await wrapper.vm.$nextTick();`.